include/envoy/request_info/request_info.h

```diff
@@ -300,6 +301,12 @@ class RequestInfo {
    * the same key overriding existing.
    */
   virtual void setDynamicMetadata(const std::string& name, const ProtobufWkt::Struct& value) PURE;
+
+  /**
+   * More general dynamic metadata object.
+   * TODO(rdsmith): Replace uses of above with this object.
+   */
+  virtual DynamicMetadata& dynamicMetadata2() PURE;
```

Comment:
Perhaps leave this out for now and replace the appropriate methods when you're ready to swap them in?

Reply:
Done.